File: remote/src/main/java/com/kelsos/mbrc/presenters/MainViewPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.ui.views.MainView;

public interface MainViewPresenter {
  void bind(MainView mainView);

  void onPause();

  void onResume();

  void onPlayPausePressed();

  void onPreviousPressed();

  void onNextPressed();

  void onStopPressed();

  void onMutePressed();

  void onShufflePressed();

  void onRepeatPressed();

  void onVolumeChange(int volume);

  void onPositionChange(int position);

  void onScrobbleToggle();

  void onLfmLoveToggle();
}

File: remote/src/main/kotlin/com/kelsos/mbrc/presenters/MainViewPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.ui.views.MainView

interface MainViewPresenter {
  fun bind(mainView: MainView)

  fun onPause()

  fun onResume()

  fun onPlayPausePressed()

  fun onPreviousPressed()

  fun onNextPressed()

  fun onStopPressed()

  fun onMutePressed()

  fun onShufflePressed()

  fun onRepeatPressed()

  fun onVolumeChange(volume: Int)

  fun onPositionChange(position: Int)

  fun onScrobbleToggle()

  fun onLfmLoveToggle()
}
